src/BSON/Decimal128.c

```diff
@@ -300,6 +300,34 @@ static phongo_create_object_retval php_phongo_decimal128_create_object(zend_clas
 #endif
 } /* }}} */
 
+static phongo_create_object_retval php_phongo_decimal128_clone_object(zval* object TSRMLS_DC) /* {{{ */
+{
+	php_phongo_decimal128_t* intern;
+	php_phongo_decimal128_t* new_intern;
+	phongo_create_object_retval new_object;
+
+	intern = Z_DECIMAL128_OBJ_P(object);
+	new_object = php_phongo_decimal128_create_object(Z_OBJCE_P(object) TSRMLS_CC);
+
+#if PHP_VERSION_ID >= 70000
+	new_intern = Z_OBJ_DECIMAL128(new_object);
+	zend_objects_clone_members(&new_intern->std, &intern->std TSRMLS_CC);
+#else
+	{
+		zend_object_handle handle = Z_OBJ_HANDLE_P(object);
+
+		new_intern = (php_phongo_decimal128_t*) zend_object_store_get_object_by_handle(new_object.handle TSRMLS_CC);
+		zend_objects_clone_members(&new_intern->std, new_object, &intern->std, handle TSRMLS_CC);
+	}
+#endif
+
+	// Use memcpy to copy bson value to avoid converting to string and back
+	memcpy(&new_intern->decimal, &intern->decimal, sizeof(bson_decimal128_t));
+	new_intern->initialized = true;
```

Comment:
Good catch.

Reply:
Thanks. This is one of the few instances where I didn't use the object initialisers to avoid converting values back and forth, at the downside of having the initialisation logic in multiple places. I think it's fine here, but in other cases we may not want to do this.